### PR TITLE
DEV: replace list control nav dropdown  with DMenuMobile

### PR DESCRIPTION
**Pull request by chapoi**
This commit:
* updates the list control navigation on mobile to use the new DMenu component
* converts the NavigationBar to glimmer
* extracts the mobile list control SCSS into its own file

### Before
CleanShot 2024-08-13 at 11 10 13@2x

### After
CleanShot 2024-08-13 at 10 59 12@2x

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/navigation-bar.gjs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+export default class NavigationBarComponent extends Component {
+  @service site;
+
+  get filterType() {
+    return filterTypeForMode(this.args.filterMode);
+  }
+
+  get selectedNavItem() {
+    const { navItems, connectors, category } = this.args;
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/components/navigation-bar.gjs:21]: I know you just ported it as-is... but all this `connectors` logic seems very weird/outdated. The only place we use `<NavigationBar` in core is here:

https://github.com/discourse/discourse/blob/9641835743899e19c9a45f129d5c69c8e4d3b922/app/assets/javascripts/discourse/app/components/d-navigation.hbs#L21C3-L26C5

And `@connectors` is not passed.

So... I think we can just drop any logic which is looking at `args.connectors`. What do you think @chapoi @renato?

Comment (chapoi) [reply app/assets/javascripts/discourse/app/components/navigation-bar.gjs:21]: I was **just** about to say the same thing, remarkable

File: app/assets/javascripts/discourse/app/components/navigation-bar.gjs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        ) {
+          let path = c.connectorClass.path(category);
+          if (path.includes(this.filterType)) {
+            item = {
+              displayName: c.connectorClass.displayName(),
+            };
+          }
+        }
+      });
+    }
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/components/navigation-bar.gjs:41]: We can burn this
```suggestion
```

File: app/assets/javascripts/discourse/app/components/navigation-bar.gjs (hunk 1 above)
Comment (davidtaylorhq) [reply app/assets/javascripts/discourse/app/components/navigation-bar.gjs:21]: I found the culprit: https://github.com/discourse/discourse/commit/2bcb5360b602f1672bf73a848c22c6d42f08309d

So we can remove connectors here

```suggestion
    const { navItems, category } = this.args;
```

(and delete the block in my other comment)

**Pull request merged**